src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -272,6 +247,60 @@ void AppHost::_HandleCreateWindow(const HWND hwnd, RECT proposedRect, winrt::Ter
         const auto nonClientSize = _window->GetTotalNonClientExclusiveSize(dpix);
         adjustedWidth = islandWidth + nonClientSize.cx;
         adjustedHeight = islandHeight + nonClientSize.cy;
+
+        const COORD origin{ gsl::narrow<short>(proposedRect.left),
+                            gsl::narrow<short>(proposedRect.top) };
+        const COORD dimensions{ Utils::ClampToShortMax(adjustedWidth, 1),
+                                Utils::ClampToShortMax(adjustedHeight, 1) };
+
+        const auto adjustedPos = Viewport::FromDimensions(origin, dimensions);
+
+        // We need to check if all corners of the window is within any screen

```

Comment:
perhaps it'd be better to GetMonitorInfo(MonitorFromWindow()) and check each side of the proposed rect with the work area edge (nudging back onscreen).